pkg/controller/cassandradatacenter/controller/controller.go

```diff
@@ -26,27 +26,25 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
-
 	// Create a new controller
 	c, err := controller.New("cassandradatacenter-controller", mgr, controller.Options{Reconciler: r})
 	if err != nil {
 		return err
 	}
 
 	// Watch for changes to primary resource CassandraDataCenter
-	err = c.Watch(&source.Kind{Type: &cassandraoperatorv1alpha1.CassandraDataCenter{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
+	if err = c.Watch(&source.Kind{Type: &cassandraoperatorv1alpha1.CassandraDataCenter{}}, &handler.EnqueueRequestForObject{}); err != nil {
 		return err
 	}
 
 	// Watch for changes to secondary resources and requeue the owner CassandraDataCenter
+	requestForOwnerHandler := &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &cassandraoperatorv1alpha1.CassandraDataCenter{},
+	}
 	for _, t := range []runtime.Object{&corev1.Service{}, &v1beta2.StatefulSet{}, &corev1.ConfigMap{}} {
-		err = c.Watch(&source.Kind{Type: t}, &handler.EnqueueRequestForOwner{
-			IsController: true,
-			OwnerType:    &cassandraoperatorv1alpha1.CassandraDataCenter{},
-		})
-		if err != nil {
-			return err
+		if err = c.Watch(&source.Kind{Type: t}, requestForOwnerHandler); err != nil {
```

Comment:
interesting, can we really reuse it just like that?

Reply:
yes, we can. `err` is just a var that's already been initialised once.

Reply:
by my concern is about the "semantics" of that, is not that somehow initialised / used differently subsequently? There is on more field in that struct and I am not sure we can just iterate over that and reuse it there ...